pkg/clusteragent/admission/patch/patcher.go

```diff
@@ -95,8 +98,11 @@ func (p *patcher) patchDeployment(req PatchRequest) error {
 		return fmt.Errorf("failed to build the JSON patch: %v", err)
 	}
 	log.Infof("Patching %s with patch %s", req.K8sTarget, string(patch))
-	_, err = p.k8sClient.AppsV1().Deployments(req.K8sTarget.Namespace).Patch(context.TODO(), req.K8sTarget.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{})
-	return err
+	if _, err = p.k8sClient.AppsV1().Deployments(req.K8sTarget.Namespace).Patch(context.TODO(), req.K8sTarget.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{}); err != nil {
+		return err
+	}
+	metrics.PatchCompleted.Inc()
```

Comment:
nitpick: no strong preference but, would it make sense to move this update to the `for` above so we have all the related metrics together?

Reply:
Yes I considered it, but this method can return a nil error without doing the patch (e.g when the config is already applied). We don't want to consider that case `completed` since the patch hasn't been performed, although it's not an error either.
I thought I'd put `metrics.PatchCompleted.Inc()` here after the k8s patch call to make sure it's accurate.